### DISCUSSION TITLE
How to troubleshoot minification error?

### DISCUSSION THREAD

**Discussion by gschuager** [webpack/webpack#14901] [General]
I'm getting a runtime error when running my app with minification enabled:

![image](https://user-images.githubusercontent.com/139658/144694488-aefbdde8-1bd5-4b5e-bee5-7ec0a73b6046.png)

```
TypeError: Y is not a function or its return value is not iterable
    at k (common.js:formatted:18)
    at o (common.js:formatted:4510)
    at m (9645.js:1)
    at Object.28882 (9645.js:1)
    at y (main.js:1)
    at Module.99645 (9645.js:1)
    at Function.y (main.js:1)
    at nt.invoke (polyfills.js:1)
    at At.run (polyfills.js:1)
    at polyfills.js:1
```

The error goes away when I set `optimization.minimize` to false.

I'm using NX + Angular 13 + module federation... I tried to create a minimal reproduction but so far I couldn't make it fail outside of my project.

Any idea how can I troubleshoot the issue?

**Comment by alexander-akait** [comment]
What is minimized do you use? I see angular uses terser and esbuild together, I am not sure which tool broke your code, maybe better to open an issue in angular repo

**Reply by gschuager** [reply]
Thank you, I didn't know that angular uses esbuild besides terser; I see the problem while toggling terser on and off ... I'm still troubleshooting, I will post an update once I come up with a conclusion.

**Reply by gschuager** [reply]
I finally came up with a reproduction: https://github.com/gschuager/mf-angular-13/tree/production-build-issue (branch production-build-issue)

But I still cannot figure out what component is to blame :)

To reproduce it, install deps with yarn and run `ng serve shell --configuration=production`. The error does not always show up, so a couple of refreshes may be required.
![image](https://user-images.githubusercontent.com/139658/145205096-5d4d2392-f525-48d2-b0ce-4d0be724374b.png)

Any input would be very appreciated.

**Reply by alexander-akait** [reply]
I can't run due:
```
shell  | An unhandled exception occurred: Cannot find module 'ngx-build-plus/package.json'
shell  | Require stack:
```

Anyway, does it happens only in `production` mode? Does you have problems if you use `optimization.minimize: false`?

**Reply by gschuager** [reply]
Thank you and sorry for the inconvenience, I just pushed a new commit fixing that issue ... you should now be able to run it with `npm start`

The runtime issue does not appear when setting minimize = false.

**Reply by alexander-akait** [reply]
I see... I think better to open an issue in angular repo, there are a lot of configurations around MF in angular, based on output, something was tree sheakabale and terser remove unused code, I'm afraid that searching and debugging everything will take me many time, it would be better if the one who working on MF in angular looked at this, you can ping me in angular repo, there are a lot of developers with whom we work together in angular repo

**Reply by gschuager** [reply]
Thank you. I already posted this info in https://github.com/angular-architects/module-federation-plugin/issues/96 but I'm not sure yet about creating an issue in angular's repo given that their tooling do not natively suport MF.

**Comment by ManAnRuck** [comment]
Hi @gschuager,

i have now the same issue with NextJS did you found a solution?
